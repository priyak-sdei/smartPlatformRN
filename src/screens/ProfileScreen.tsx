import React from 'react';
import { View, Text, StyleSheet, Button } from 'react-native';
import { useAppSelector, useAppDispatch } from '../redux/hooks';
import { clearUser } from '../redux/slices/userSlice';

const ProfileScreen = () => {
    const email = useAppSelector(state => state.user.email);
    const dispatch = useAppDispatch();

    const handleLogout = () => {
        dispatch(clearUser());
    };

    return (
        <View style={styles.container} testID="profile-root">
            <Text style={styles.title}>Profile Screen</Text>
            {email && <Text style={styles.email}>Email: {email}</Text>}
            <Button title="Logout" onPress={handleLogout} />
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    title: {
        fontSize: 24,
        fontWeight: 'bold',
    },
    email: {
        marginVertical: 16,
        fontSize: 16,
    },
});

export default ProfileScreen; 