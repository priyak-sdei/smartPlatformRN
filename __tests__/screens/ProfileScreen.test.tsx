import React from 'react';
import { render, waitFor, cleanup } from '@testing-library/react-native';
import ProfileScreen from '../../src/screens/ProfileScreen';
import { Provider } from 'react-redux';
import { store } from '../../src/redux/store';
import renderer, { act } from 'react-test-renderer';

beforeAll(() => {
    // Save the original console.error
    // @ts-ignore
    console._errorOriginal = console.error;
    jest.spyOn(console, 'error').mockImplementation((msg) => {
        if (
            typeof msg === 'string' &&
            msg.includes('inside a test was not wrapped in act')
        ) {
            return;
        }
        // @ts-ignore
        console._errorOriginal(msg);
    });
});

afterAll(() => {
    // @ts-ignore
    if (console._errorOriginal) {
        // @ts-ignore
        console.error = console._errorOriginal;
    }
});

afterEach(cleanup);

describe('ProfileScreen', () => {
    it('renders without crashing', async () => {
        const { getByTestId } = render(
            <Provider store={store}>
                <ProfileScreen />
            </Provider>
        );
        await waitFor(() => {
            expect(getByTestId('profile-root')).toBeTruthy();
        });
    });

    it('matches snapshot', () => {
        let tree: renderer.ReactTestRenderer | null = null;
        act(() => {
            tree = renderer.create(
                <Provider store={store}>
                    <ProfileScreen />
                </Provider>
            );
        });
        expect(tree!.toJSON()).toMatchSnapshot('ProfileScreen component');
    });
}); 