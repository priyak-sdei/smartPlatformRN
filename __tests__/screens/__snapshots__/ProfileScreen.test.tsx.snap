// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen matches snapshot: ProfileScreen component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="profile-root"
>
  <Text
    style={
      {
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Profile Screen
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Logout
      </Text>
    </View>
  </View>
</View>
`;
